crates/rolldown/src/ast_scanner/impl_visit.rs

```diff
@@ -47,30 +81,55 @@ impl<'me, 'ast> Visit<'ast> for AstScanner<'me> {
     self.try_diagnostic_forbid_const_assign(symbol_id);
   }
 
-  fn visit_identifier_reference(&mut self, ident: &IdentifierReference) {
-    let symbol_id = self.resolve_symbol_from_reference(ident);
-    match symbol_id {
-      Some(symbol_id) if self.is_top_level(symbol_id) => {
-        self.add_referenced_symbol(symbol_id);
+  fn visit_member_expression(&mut self, expr: &MemberExpression<'ast>) {
```

Comment:
Do we need to call `walk::walk_member_expression` here?

Reply:
No, if you call `walk::walk_member_expresion`, then `identifier` visitor will be called, and another referenced symbol will be added to this module. Visitor of  `member_expr_reference` and `identifier_reference` should be mutually exclusive

Reply:
I could add `enter_node` and `leave_node` if we need.

Reply:
I just look it again. Could you rewrite it like

```rust
fn visit_member_expression{
  if MemberExpression::StaticMemberExpression ... {
    // ...
    // Bailout for not creating another referenced symbol
    return
  }

  // nothing special, just keep walking
  walk::visit_member_expression(self)
}
```

The current logic is complex to see when we should keep walking when we should bailout. The other visitor's function is writing like this way. It would be good if we could keep this style.

- https://github.com/rolldown/rolldown/blob/aef45e09fa93e7df8f18422fb8d5c47fae0c77f0/crates/rolldown/src/module_finalizers/scope_hoisting/impl_visit_mut.rs#L356-L360
- https://github.com/rolldown/rolldown/blob/aef45e09fa93e7df8f18422fb8d5c47fae0c77f0/crates/rolldown/src/module_finalizers/scope_hoisting/impl_visit_mut.rs#L421-L464
- https://github.com/rolldown/rolldown/blob/aef45e09fa93e7df8f18422fb8d5c47fae0c77f0/crates/rolldown/src/ast_scanner/impl_visit.rs#L74-L79